packages/components/src/internal/components/editable/actions.ts

```diff
@@ -457,7 +460,11 @@ async function convertRowToEditorModelData(
             message = messageAndValue.message;
         }
     } else {
-        valueDescriptors = valueDescriptors.push({ display: data, raw: data });
+        let display = data;
+        if (col.isTimeOrDateTimeColumn && typeof data === 'string') {
```

Comment:
The other if statement is defensive against `col=undefined`. Let's do the same here:

```tsx
if (col?.isTimeOrDateTimeColumn && typeof data === 'string') {
    display = getDateTimeDisplayValueFromStr(data, col);
}
```